src/util/validation/zod.ts

```diff
@@ -37,6 +37,8 @@ const boolStr = z.union([z.literal('true'), z.literal('false')])
 const url = z.string().url('Must be a valid URL')
 const positiveNumStrRegex = /^[+]?([.]\d+|\d+[.]?\d*)$/
 const amount = z.string().regex(positiveNumStrRegex, 'Must be a positive number')
+const denomStrRegex = /^[a-zA-Z][a-zA-Z0-9/:._-]{2,127}$/
+const denom = z.string().regex(denomStrRegex, 'Must be a valid denomination')
```

Comment:
This is not necessary at the moment but will be valuable when we add support for selected denoms.